### PR TITLE
Db migration

### PR DESCRIPTION
**Pull request by SmetjukVlad**
Created migrations for databases based on the module YOYO

### CODE REVIEW

File: config/telegram.env.dist [hunk 1]
```diff
@@ -1,2 +1,8 @@
 TELEGRAM_API_TOKEN=""
 TELEGRAM_ADMIN_ID=""
+PATH_SOURCES="./migrations"
```
Comment (alexby) [inline config/telegram.env.dist]: Please change the name to something more specific. E.g. `MIGRATIONS_PATH`

File: config/telegram.env.dist [hunk 2]
```diff
@@ -1,2 +1,8 @@
...
+DB_LOGIN="postgres"
+DB_PASSWORD="postgres"
+CONTAINER_NAME="db"
```
Comment (alexby) [inline config/telegram.env.dist]: It's not "container name", it's a "database host". The same about `CONTAINER_PORT`.

File: config/telegram.env.dist [hunk 3]
```diff
@@ -1,2 +1,8 @@
...
+CONTAINER_PORT="5432"
+DB_NAME="postgres"
```
Comment (alexby) [inline config/telegram.env.dist]: Here should be an empty line in the end of the file

File: migrations/20200730_02_create_table_transactions.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+         "created_by integer,"
+         "created_at date,"
+         "amount integer,"
+         "creditor_id integer,"
+         "debtor_id integer,"
+         "FOREIGN KEY(created_by) REFERENCES users(id),"
+         "FOREIGN KEY(creditor_id) REFERENCES users(id),"
+         "FOREIGN KEY(debtor_id) REFERENCES users(id))",
+         "DROP TABLE transactions")
+]
```
Comment (alexby) [inline migrations/20200730_02_create_table_transactions.py]: Here should be an empty line in the end of the file

File: migrations/20200730_01_create_table_users.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from yoyo import step
+
+__depends__ = {}
+
+steps = [
+    step("CREATE TABLE users (id integer PRIMARY KEY,"
```
Comment (alexby) [inline migrations/20200730_01_create_table_users.py]: Can you please remind me where did we have a documentation about this structure?

File: migrations/20200730_02_create_table_transactions.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+from yoyo import step
+
+__depends__ = {'20200730_01_create_table_users'}
```
Comment (alexby) [inline migrations/20200730_02_create_table_transactions.py:3]: Why is it required?

File: requirements.txt [hunk 7]
```diff
@@ -3,3 +3,6 @@ argh===0.26.2
 psycopg2===2.8.4
 pyyaml===5.3.1
 watchdog===0.10.2
+requests===2.23.0
```
Comment (alexby) [inline requirements.txt]: Why do we need it?

File: requirements.txt [hunk 8]
```diff
@@ -3,3 +3,6 @@ argh===0.26.2
...
+datetime===4.3
```
Comment (alexby) [inline requirements.txt]: Why do we need it?

File: yoyo.ini [hunk 9]
```diff
@@ -0,0 +1,8 @@
+[DEFAULT]
+sources = %(PATH_SOURCES)s
+database = postgresql://%(DB_LOGIN)s:%(DB_PASSWORD)s@%(CONTAINER_NAME)s:%(CONTAINER_PORT)s/%(DB_NAME)s
+migration_table = _yoyo_migration
+batch_mode = on
+verbosity = 1
+post_create_command = hg add {}
```
Comment (alexby) [inline yoyo.ini]: Why do we need it?

**Review by alexby — CHANGES_REQUESTED** [review]

File: migrations/20200730_02_create_table_transactions.py (hunk 6 above)
Comment (SmetjukVlad) [reply migrations/20200730_02_create_table_transactions.py:3]: in this module you can create a dependency to undo migrations one at a time, if needed.
therefore, in each new file, the previous file is written.

File: requirements.txt (hunk 7 above)
Comment (SmetjukVlad) [reply requirements.txt]: really don’t, I will take it away.

File: requirements.txt (hunk 8 above)
Comment (SmetjukVlad) [reply requirements.txt]: really don’t, I will take it away

File: yoyo.ini (hunk 9 above)
Comment (SmetjukVlad) [reply yoyo.ini]: I haven't figured it out completely, I'll write an answer later.

File: config/telegram.env.dist (hunk 1 above)
Comment (SmetjukVlad) [reply config/telegram.env.dist]: fixed it

File: config/telegram.env.dist (hunk 2 above)
Comment (SmetjukVlad) [reply config/telegram.env.dist]: fixed it

File: config/telegram.env.dist (hunk 3 above)
Comment (SmetjukVlad) [reply config/telegram.env.dist]: fixed it

File: migrations/20200730_02_create_table_transactions.py (hunk 4 above)
Comment (SmetjukVlad) [reply migrations/20200730_02_create_table_transactions.py]: fixed it

File: config/telegram.env.dist (hunk 1 above)
Comment (alexby) [reply config/telegram.env.dist]: I don't see the changes

Comment (alexby) [reply config/telegram.env.dist]: It looks like you forgot to push your changes. I see the same below

File: config/telegram.env.dist (hunk 3 above)
Comment (alexby) [reply config/telegram.env.dist]: what about it?

File: app/db/createdb.sql  [hunk 10]
```diff
@@ -19,4 +19,4 @@ create table transactions
     FOREIGN KEY(created_by) REFERENCES users(id),
     FOREIGN KEY(creditor_id) REFERENCES users(id),
     FOREIGN KEY(debtor_id) REFERENCES users(id)
-);
+);
```
Comment (alexby) [inline app/db/createdb.sql ]: Why is this change here?

File: config/telegram.env.dist [hunk 11]
```diff
@@ -1,2 +1,8 @@
 TELEGRAM_API_TOKEN=""
 TELEGRAM_ADMIN_ID=""
+MIGRATIONS_PATH="./migrations"
+DB_LOGIN="postgres"
```
Comment (alexby) [inline config/telegram.env.dist]: The names should be consistent - if you have something named `DB_*` you cannot have name `DATABASE_*` and vise versa.

File: yoyo.ini (hunk 9 above)
Comment (alexby) [reply yoyo.ini]: And where is the answer?

**Review by alexby — CHANGES_REQUESTED** [review]

**Review by alexby — APPROVED** [review]
